apps/emqx/integration_test/emqx_persistent_session_ds_SUITE.erl

```diff
@@ -169,20 +193,57 @@ list_all_subscriptions(Node) ->
 list_all_pubranges(Node) ->
     erpc:call(Node, emqx_persistent_session_ds, list_all_pubranges, []).
 
-prop_only_cores_run_gc(CoreNodes) ->
-    {"only core nodes run gc", fun(Trace) -> ?MODULE:prop_only_cores_run_gc(Trace, CoreNodes) end}.
-prop_only_cores_run_gc(Trace, CoreNodes) ->
+prop_only_cores_run_session_gc(CoreNodes) ->
+    {"only core nodes run session gc", fun(Trace) ->
+        ?MODULE:prop_only_cores_run_session_gc(Trace, CoreNodes)
+    end}.
+prop_only_cores_run_session_gc(Trace, CoreNodes) ->
+    prop_only_cores_run_gc(Trace, CoreNodes, session).
+
+prop_only_cores_run_message_gc(CoreNodes) ->
+    {"only core nodes run message gc", fun(Trace) ->
+        ?MODULE:prop_only_cores_run_message_gc(Trace, CoreNodes)
+    end}.
+prop_only_cores_run_message_gc(Trace, CoreNodes) ->
+    prop_only_cores_run_gc(Trace, CoreNodes, message).
+
+prop_only_cores_run_gc(Trace, CoreNodes, Kind) ->
+    Tags =
+        case Kind of
+            session ->
+                [ds_session_gc, ds_session_gc_lock_taken];
+            message ->
+                [ds_message_gc, ds_message_gc_lock_taken]
+        end,
     GCNodes = lists:usort([
         N
      || #{
             ?snk_kind := K,
             ?snk_meta := #{node := N}
         } <- Trace,
-        lists:member(K, [ds_session_gc, ds_session_gc_lock_taken]),
+        lists:member(K, Tags),
         N =/= node()
     ]),
     ?assertEqual(lists:usort(CoreNodes), GCNodes).
 
+assert_drain_expected(ExpectedMessages, Opts) ->
+    Timeout = maps:get(timeout, Opts, 2_500),
```

Comment:
A set.  The order shouldn't matter, because we're interested in _eventually_ seeing this set. 

2,5 s seems ok given an idle poll interval of 100 ms and half a dozen messages being published.

Reply:
This function looks like something that can be reused by other tests. So it would be nice to have a description and a typespec to formalize that it exactly it checks. It would be a tremendous help in case the testcase starts failing for any reason.